JobServiceTest.testServiceRecursive test is failing on Python 2 for no apparent reason

See #2880 and #2877. We get a failure like this:

```
_____________________ JobServiceTest.testServiceRecursive ______________________
Traceback (most recent call last):
  File "/builds/databiosphere/toil/src/toil/test/src/jobServiceTest.py", line 133, in testServiceRecursive
    self.runToil(t)
  File "/builds/databiosphere/toil/src/toil/test/src/jobServiceTest.py", line 185, in runToil
    self.fail() #Exceeded a reasonable number of restarts
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: None
```
The log at https://ucsc-ci.com/databiosphere/toil/-/jobs/31750/raw makes it look like maybe our badWorker is just failing things too much? But the service itself isn't working right and shipping numbers around because it can't deal with the initial empty file.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-467)
┆Issue Number: TOIL-467

--- a/src/toil/test/src/jobServiceTest.py
+++ b/src/toil/test/src/jobServiceTest.py
@@ -18,6 +18,7 @@ from builtins import range
 import codecs
 import os
 import random
+import traceback
 import unittest
 
 # Python 3 compatibility imports
@@ -277,7 +278,7 @@ class ToyService(Job.Service):
         try:
             while True:
                 if terminate.isSet(): # Quit if we've got the terminate signal
-                    logger.debug("Demo service worker being told to quit")
+                    logger.debug("Service worker being told to quit")
                     return
 
                 time.sleep(0.2) # Sleep to avoid thrashing
@@ -288,20 +289,27 @@ class ToyService(Job.Service):
                         fH = codecs.getreader('utf-8')(fH)
                         line = fH.readline()
                 except:
-                    logger.debug("Something went wrong reading a line")
+                    logger.debug("Something went wrong reading a line: %s", traceback.format_exc())
                     raise
 
+                if len(line.strip()) == 0:
+                    # Don't try and make an integer out of nothing
+                    continue
+                    
                 # Try converting the input line into an integer
                 try:
                     inputInt = int(line)
                 except ValueError:
-                    logger.debug("Tried casting input line to integer but got error: %s" % line)
+                    logger.debug("Tried casting input line '%s' to integer but got error: %s", line, traceback.format_exc())
                     continue
 
                 # Write out the resulting read integer and the message              
                 with jobStore.updateFileStream(outJobStoreID) as fH:
                     fH.write(("%s %s\n" % (inputInt, messageInt)).encode('utf-8'))
+                    
+                logger.debug("Service worker did useful work")
         except:
+            logger.debug("Error in service worker: %s", traceback.format_exc())
             error.set()
             raise
 

